File: RB101_Programming_Foundations/Small Probems/Easy7/multiply_lists.rb
# input: two arrays
# output : an array with each element the product of the two array's respective index value

# iterate through the array, find the product, store it in a new array

# def multiply_list(array1, array2)
#   element_numbers = array1.size
#   product_array = Array.new(element_numbers)

#   element_numbers.times do |idx|
#     product_array[idx] = array1[idx] * array2[idx]
#   end

#   product_array
# end

def multiply_list(array1, array2)
  array1.zip(array2).map {|arr| arr[0] * arr[1] }
end




p multiply_list([3, 5, 7], [9, 10, 11]) # == [27, 50, 77]

File: js210_fundamentals_of_JS_Programming/problem_sets/small_problems/easy4/multiply_lists.js
function multiplyList(array1, array2) {
  let result = [];

  for (let index = 0; index < array1.length; index += 1) {
    result[index] = array1[index] * array2[index];
  }
  return result;
}

console.log(multiplyList([3, 5, 7], [9, 10, 11]));    // [27, 50, 77]
